account_payment_term_partner_holiday/models/account_invoice.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.onchange("date_due")
+    def _onchange_date_due(self):
+        if self.date_due and self.payment_term_id and self.partner_id:
+            new_date_due = self.date_due
+            is_date_in_holiday = self.partner_id.is_date_in_holiday_ids(new_date_due)
+            pterm = self.payment_term_id.with_context(
+                currency_id=self.company_id.currency_id.id
+            )
+            while is_date_in_holiday:
```

Comment:
Instead of this loop, I think the called method should return the forbidden interval (or False if not), and then put following day of the end date.